File: 10000/700/10775_공항/BJ_10775_공항.java
// https://www.acmicpc.net/problem/10775

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.stream.IntStream;

public class BJ_10775_공항 {
    public static class GateManager {
        private int[] gates;

        public GateManager(final int N) {
            gates = IntStream.range(0, N + 1).toArray();
        }

        public int findEmptyGate(final int g) {
            if (gates[g] != g) {
                gates[g] = findEmptyGate(gates[g]);
            }
            return gates[g];
        }

        public boolean tryToDock(final int g) {
            if (findEmptyGate(g) == 0) {
                return false;
            }

            --gates[findEmptyGate(g)];
            return true;
        }
    }

    public static void main(final String[] args) throws IOException {
        try (final var stdin = new BufferedReader(new InputStreamReader(System.in))) {
            final var G = Integer.parseInt(stdin.readLine());
            final var P = Integer.parseInt(stdin.readLine());

            var numDockings = 0;
            final var gateManager = new GateManager(G);
            for (int i = 0; i < P; ++i) {
                final var g = Integer.parseInt(stdin.readLine());
                if (!gateManager.tryToDock(g)) {
                    break;
                }

                ++numDockings;
            }

            System.out.print(numDockings + "\n");
        }
    }
}

File: 10000/700/10775_공항/BJ_10775_공항.py
# https://www.acmicpc.net/problem/10775

import sys


def find(parents, u):
    if parents[u] != u:
        parents[u] = find(parents, parents[u])
    return parents[u]


def union(parents, v, u):
    vp = find(parents, v)
    up = find(parents, u)
    parents[vp] = up


sys.setrecursionlimit(10**5)
readline = lambda: sys.stdin.readline().rstrip()  # noqa

G = int(readline())
gates = [*range(G + 1)]

P = [int(readline()) for _ in range(int(readline()))]
n_dockings = 0
for g in P:
    if find(gates, g) == 0:
        break

    gates[find(gates, g)] -= 1
    n_dockings += 1

print(n_dockings)
